.gitignore

```diff
@@ -20,3 +20,6 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+# Maven build directories
+**/target/
```

Comment:
Although I agree with this approach, it's not what is widely used and agreed on in KIE repositories. We basically use `.gitignore` in every single maven module and specify the `/target` there. Having `**/target/` in the top-level `.gitignore` will also match things like `src/main/java/my/package/target/` which will lead into ignoring source dir.